src/sock.c

```diff
@@ -99,6 +100,17 @@ sock_t sock_connect(const char * const host, const unsigned int port)
         if (sock < 0)
             continue;
 
+	/* turn on keepalive */
```

Comment:
Found that: https://msdn.microsoft.com/en-us/library/windows/desktop/ee470551%28v=vs.85%29.aspx
And that: https://msdn.microsoft.com/en-us/library/windows/desktop/dd877220%28v=vs.85%29.aspx

Seems that intervals can be set using slightly different interface on windows - WSAIoctl with SIO_KEEPALIVE_VALS, however it's not clear how to get "socket" handle for WSAIoctl from "SOCKET" handle, but it must be possible somehow, since win32 socket/connect/close emulated over Winsock interface .

What if I just add an exported function xmpp_conn_set_keepalive( timeout, interval )? User can use some values to activate keepalive and set times or just 0,0 to set keepalive off. By default, no actions will be taken, and keepalive setting will be in "system default" state.
This should not brake anything, just transparently add a feature.